pennylane/devices/tests/test_templates.py

```diff
@@ -0,0 +1,995 @@
+        expected = 1.0
+        assert np.isclose(res, expected, atol=tol(dev.shots))
+
+    def test_IQPEmbedding(self, device, tol):
+        """Test the IQPEmbedding template."""
+        dev = device(3)
+
+        @qml.qnode(dev)
+        def circuit(features):
+            qml.IQPEmbedding(features, wires=range(3), n_repeats=4)
+            return [qml.expval(qml.Z(w)) for w in range(3)]
+
+        res = circuit([1.0, 2.0, 3.0])
+        expected = [0.40712208, 0.32709118, 0.89125407]
+        assert np.allclose(res, expected, atol=tol(dev.shots))
+
+    @pytest.mark.xfail(reason="most devices do not support CV")
+    def test_Interferometer(self, device):
+        """Test the Interferometer template."""
+        dev = device(4)
+
+        @qml.qnode(dev)
+        def circuit(params):
+            qml.Interferometer(*params, wires=range(4))
+            return qml.expval(qml.Identity(0))
+
+        shapes = [[6], [6], [4]]
+        params = []
+        for shape in shapes:
+            params.append(np.random.random(shape))
+
+        _ = circuit(params)
+
+    def test_LocalHilbertSchmidt(self, device, tol):
+        """Test the LocalHilbertSchmidt template."""
+        dev = device(4)
+        u_tape = qml.tape.QuantumScript([qml.CZ(wires=(0, 1))])
+
+        def v_function(params):
+            qml.RZ(params[0], wires=2)
+            qml.RZ(params[1], wires=3)
+            qml.CNOT(wires=[2, 3])
+            qml.RZ(params[2], wires=3)
+            qml.CNOT(wires=[2, 3])
+
+        @qml.qnode(dev)
+        def local_hilbert_test(v_params, v_function, v_wires, u_tape):
+            qml.LocalHilbertSchmidt(v_params, v_function=v_function, v_wires=v_wires, u_tape=u_tape)
+            return qml.probs(u_tape.wires + v_wires)
+
+        def cost_lhst(parameters, v_function, v_wires, u_tape):
+            # pylint:disable=unsubscriptable-object
+            return (
+                1
+                - local_hilbert_test(
+                    v_params=parameters, v_function=v_function, v_wires=v_wires, u_tape=u_tape
+                )[0]
+            )
+
+        res = cost_lhst(
+            [3 * np.pi / 2, 3 * np.pi / 2, np.pi / 2],
+            v_function=v_function,
+            v_wires=[2, 3],
+            u_tape=u_tape,
+        )
+        expected = 0.5
+        assert np.isclose(res, expected, atol=tol(dev.shots))
+
+    def test_MERA(self, device, tol):
+        """Test the MERA template."""
+
+        def block(weights, wires):
+            qml.CNOT(wires=[wires[0], wires[1]])
+            qml.RY(weights[0], wires=wires[0])
+            qml.RY(weights[1], wires=wires[1])
+
+        n_wires = 4
+        n_block_wires = 2
+        n_params_block = 2
+        n_blocks = qml.MERA.get_n_blocks(range(n_wires), n_block_wires)
+        template_weights = [[0.1, -0.3]] * n_blocks
+        dev = device(n_wires)
+
+        dev = qml.device("default.qubit", wires=range(n_wires))
```

Comment:
```suggestion
```

?

Reply:
bad copy-paste from the template docstrings, sorry!